core/src/main/java/com/linecorp/armeria/server/DefaultExceptionHandler.java

```diff
@@ -45,22 +43,7 @@
 
     /**
      * Converts the specified {@link Throwable} to an {@link HttpResponse}.
-     *
-     * <p>Implementation note:
-     * A failed {@link HttpResponse} should be returned in order to let a {@link RequestLog} complete deferred
-     * values. The given cause could be raised before setting deferred values of a {@link RequestLog}.
-     * See https://github.com/line/armeria/issues/3719.
-     *
-     * <p>For example:<pre>{@code
-     * // Bad - Deferred values will not be completed.
-     * HttpResponse.of(HttpStatus.BAD_REQUEST);
-     * // Good - A LoggingService will not log any exception and complete deferred values.
-     * HttpResponse.ofFailure(HttpStatusException.of(HttpStatus.BAD_REQUEST));
-     * // Good - A LoggingService will log the IllegalStateException and complete deferred values.
-     * HttpResponse.ofFailure(HttpStatusException.of(HttpStatus.BAD_REQUEST, new IllegalStateException(...)));
-     * }</pre>
      */
-    @Nullable
```

Comment:
nit: Add `@Nonnull`?

Reply:
Should we now update the default implementation to use `HttpResponse.of(...)` except for `HttpStatusException` and `HttpStatusException`?

Reply:
Thanks, fixed. 😉